ehsm_kms_service/ehsm_kms_server.js

```diff
@@ -43,7 +49,24 @@ const _checkSign = function (req,res,next){
     res.send(result(400,'Missing required parameters'));
     return;
   }
-  // todo sign 
+  let test_app_key = '202112345678';
+
+  let str = '';
+  let sign_parmas = {appid, nonce, timestamp} 
+  for(var k in sign_parmas){
+    if(!str) {
+      str += k + '=' + sign_parmas[k]
+    } else{
+      str += '&' + k + '=' + sign_parmas[k]
+    }
+  }
+  str += '&app_key=' + test_app_key;
```

Comment:
app_key is for internal hmac-sha256 calculation only. 
you can hardcode a 32B app key in .js and .py, but not transfer btw them.